IdentityCore/src/webview/background/ios/MSIDBackgroundTaskManager.m

```diff
@@ -65,55 +66,62 @@ + (MSIDBackgroundTaskManager *)sharedInstance
 
 /*
  Background task execution:
- https://forums.developer.apple.com/message/253232#253232
+ https://developer.apple.com/forums/thread/85066
  */
 
 - (void)startOperationWithType:(MSIDBackgroundTaskType)type
 {
-    UIBackgroundTaskIdentifier backgroundTask = [self backgroundTaskWithType:type];
+    MSIDBackgroundTaskData *backgroundTaskData = [self backgroundTaskWithType:type];
```

Comment:
Seems the implementation of this class is not thread-safe?

E.g., if there are two silent calls going on at the same time. The execution sequence could be:
1. Silent call A execute line 103 of this file;
2. Silent call B execute line 74;
3. Silent call A execute line 104-105;
4. Silent call B execute line 76-80.

In the above sequence, Silent call B loses background task protection. What do you think?

Reply:
In this example, 
1. Silent call A calls method to stop bg task operation.
2. Silent call B gets reference to Bg task that was started by A from cache at line 74.
3. Silent call A ends bg task and removes bg task data object from cache. But in 2. since the bg task ref is still retained, it won't be deallocated.
4. Silent call B will not start a new bg task.

Great catch @jasoncoolmax ! I think in step 2. when silent call B gets reference to bg task data, if I mark it as a weak reference, that might resolve this. What do you think?

Reply:
I am fine with the weak reference solution, given that using @ synchronized may sacrifice some performance here.

Reply:
@jasoncoolmax Added test case to test if network requests might loose background task protection in the event that multiple nw requests simultaneously call for bg task protection through threads. I think OneDrive/Office does this to refresh tokens in its cache periodically via multiple threads.

The test case was failing without synchronized block for start & stop bg task operation methods. I've added synchronized blocks per your suggestion to both methods so that no 2 threads enter those methods simultaneously.
Please let me know if I missed something :)